### DISCUSSION TITLE
Elfinder rename, copy, move operations stuck, keep showing progress popup with Google Cloud integration

### DISCUSSION THREAD

**Discussion by devDev1922** [Studio-42/elFinder#3358] [Q&A]
I am trying to integrate Google Cloud bucket with Elfinder. Operations like `rename`, `copy`, `move` stuck if again try to perform any one of the operation. For example, for the first time if I rename a file then it works, if I try to again `rename`, `copy`, `move` the file then the progress popup keeps showing until refresh the page again. 

I also noticed selecting file one after one creates the same issue. For example, I have 4 files in a google cloud folder like abc.txt, def.txt, ghi.png, jkl.jpg. After Elfinder loads completely, if I select abc.txt and then without performing any operation if I again select another file lets say jkl.png then perform the rename operation then again it stuck and stops at 80%. 

Here is the integration code:

                            $client = new StorageClient([
					'keyFilePath' => $file_path, 
				]);
				$bucket = $client->bucket($bucket_name);
				
				$endpoint = 'https://storage.googleapis.com/'. $bucket_name;
				
				$gc_adapter = new GoogleStorageAdapter($client, $bucket);
				
				$gc_filesystem = new Filesystem($gc_adapter , Array ( "url" => $endpoint ));
				
                $root_options = array(
                    'debug' => true,
                    'roots' => [
                
                        [
                            'driver' => 'Flysystem',
                            'filesystem' => $gc_filesystem
                        ]
                
                    ]
                );
                
                $connector = new elFinderConnector(new elFinder($root_options));
                $connector->run();

I noticed that when the operation works fine, then there are 2 ajax request one for `cmd: url` and after that `cmd: rename`. In case when operation stuck at that time there is only one request of `cmd: url`.

I also posted the same issue on [StackOverflow](https://stackoverflow.com/questions/67673025/elfinder-rename-copy-move-operations-stuck-keep-showing-progress-popup-with-g)

**Comment by paulcanning** [comment]
Are you able to step through the code with XDebug? Or try to log out some stuff, see if you can spot an error?

**Comment by devDev1922** [comment]
Earlier I didn't checked by enabling the debug of elfinder. I just enabled it and getting message of `Driver "elFinderVolume" does not exist`. But local filesystem is working fine. I am only facing issue with google cloud addon
